### ISSUE TITLE
https://github.com/Praqma/c3po-dsl - anything to save from there before I kill the repo?

### ISSUE THREAD

**Issue by buep** [Praqma/JenkinsAsCodeReference#120]
Once upon a time @praqma-thi spent an amount of time on a setup to spin of Jenkins easily to test plugins etc.

It never finished, but only lives still because there might be something jenkins as code want to keep?
See this internal discussion: https://praqma.fogbugz.com/f/cases/13800/Spin-up-a-Jenkins-plugin-playground-was-Set-up-Prestested-Integration-test-environment

It might not be script, but could just be a use-case you yet doesn't cover and want to create an issue on.

Please have a look on the internal discussion and the repo: https://github.com/Praqma/c3po-dsl

Ping me back @Andrey9kin when done - I will delete the c3po-dsl if there is nothing to save.

**Comment by praqma-thi** [comment]
C3PO relied on a gimmick that allowed you to generate a jenkins.war file that came with a specific plugin and all its dependencies preinstalled. That war file was dropped in a Docker image and a container was spun up. It allowed you to test a specific commit of any Jenkins plugin that had a dependency to the maven-hpi-plugin. 

It had some potential for automation/testing as it's basically an automatic hpi:run, which spins the jenkins setup up locally. It never got anywhere though, as it doesn't cover many scenarios that local hpi:runs or functional testing with the jenkins harness don't already.

**Comment by praqma-thi** [comment]
TL;DR It was a fun exercise, but nothing will be lost if the repositories are deleted, other than good memories over time, I suppose.

**Comment by Andrey9kin** [comment]
@buep probably not

**Comment by buep** [comment]
I deleted the Praqma/c3po-dsl repository.
